### DISCUSSION TITLE
Support of keras/TF models?

### DISCUSSION THREAD

**Discussion by andreapago** [MAIF/shapash#301] [Q&A]
Hello Shapash team,

I am investigating the use of shapash to make my work more explainable. I am using keras/TF to create my model, and it seems the model (type) is not supported. Could you please confirm that? since I found not much documentation about that.
Furthermore, are you planning to support it in the future?

Many thanks.

**Comment by yg79** [comment]
Currently Shapash cannot be used with TF models.
Do you do NLP, Computer Vision or supervised ML on Tabular Dataset?

**Reply by andreapago** [reply]
That's a pity...I actually work on seismological data applying CNN (originally) and graph attention networks (lately). These data are multi-dimensional and actually it is not really straightforward to have them in a meaningful tabular way...

**Comment by yg79** [comment]
TF on tabular dataset could easily explainable in Shapash. We will work on it soon with the emergence of models such as Tabnet and others.
You are welcome to work on this issue with us.

**Reply by andreapago** [reply]
Let's say I manage to have a meaningful tabular way of representing my seismological data, which are the classes that are involved for adding keras/TF models to the explainability in shapash? Could you guide me a bit so I can look and maybe see what I could do.
Thanks in advance.

**Reply by yg79** [reply]
I think you can use this tutorial: https://github.com/MAIF/shapash/blob/master/tutorial/explainer/tuto-expl01-Shapash-Viz-using-Shap-contributions.ipynb
--> you can use shap kernel or deep explainer to compute your shapley values and use shapash visualisations
